lib/bundler/fetcher/downloader.rb

```diff
@@ -14,7 +14,7 @@ def fetch(uri, options = {}, counter = 0)
         raise HTTPError, "Too many redirects" if counter >= redirect_limit
 
         response = request(uri, options)
-        Bundler.ui.debug("HTTP #{response.code} #{response.message}")
+        Bundler.ui.debug("HTTP #{response.code} #{response.message} #{uri}")
```

Comment:
Oh, heh.  Yeah, usually the message was something like "Not Modified", so the full thing was just "HTTP 304 Not Modified" (without the URI of course).